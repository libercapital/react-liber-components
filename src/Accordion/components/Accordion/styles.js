import styled from 'styled-components';
import { media } from '../../../Util';
import { handleThemeFromObject } from '../../../Themes';

export const AccordionBox = styled.div`
  overflow: hidden;
  width: 100%;
  background-color: ${props =>
    props.open
      ? handleThemeFromObject(props.theme, 'accordion.boxBackgroundOpen', '#fff')
      : handleThemeFromObject(props.theme, 'accordion.boxBackgroundClosed', '#f9f9f9')};
  border: ${props =>
    props.open
      ? `1px solid ${handleThemeFromObject(props.theme, 'accordion.boxBorderOpen', '#fff')}`
      : `solid 1px ${handleThemeFromObject(
          props.theme,
          'accordion.boxBorderClosed',
          'rgba(150, 150, 150, 0.18)',
        )}`};
  transition: all 400ms ${props => (props.open ? 'ease-in' : 'ease-out')};
  &:hover {
    background-color: ${props =>
      props.open
        ? handleThemeFromObject(props.theme, 'accordion.boxBackgroundOpen', '#fff')
        : handleThemeFromObject(props.theme, 'accordion.boxBackgroundHover', '#f3f3f3')};
  }
`;

export const ShowBox = styled.div`
  max-height: ${props => (props.open ? props.size : '0px')};
  transition: max-height 400ms ${props => (props.open ? 'ease-in' : 'ease-out')};
  overflow: hidden;
`;

export const AccordionAction = styled.div`
  padding: 20px 50px;
  cursor: pointer;
  box-sizing: border-box;
  display: inline-block;
  background-color: ${props => (props.open ? props.color || 'transparent' : 'transparent')};
  width: 100%;
  transition: all 400ms ${props => (props.open ? 'ease-in' : 'ease-out')};
  position: relative;

  ${media.large`
    padding: 30px 50px;
  `}

  &:after {
    position: absolute;
    top: calc(50% - 4px);
    left: 15px;
    width: 0;
    height: 0;
    padding: 0;
    content: '';
    border-left: 7px solid transparent;
    border-right: 7px solid transparent;
    border-top: 8px solid
      ${props =>
        props.open
          ? handleThemeFromObject(props.theme, 'accordion.actionBorderTopOpen', '#009dff')
          : handleThemeFromObject(props.theme, 'accordion.actionBorderTopClosed', '#4e4e4e')};
    pointer-events: none;
    transform: ${props => (props.open ? 'rotate(0deg)' : 'rotate(-90deg)')};
    transition: all 400ms ${props => (props.open ? 'ease-in' : 'ease-out')};
  }
`;
