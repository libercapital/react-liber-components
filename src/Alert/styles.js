import styled, { css } from 'styled-components';
import { MdClose } from '../Icons/MaterialDesign';

const icon = css`
  --size: 24px;
  min-width: var(--size);
  max-width: var(--size);
  min-height: var(--size);
  max-height: var(--size);
`;

export const CloseIcon = styled(MdClose)`
  ${icon}
  &:hover {
    cursor: pointer;
  }
`;

export const AlertBox = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  width: 100%;
`;

export const AlertContent = styled.div`
  display: flex;
  flex-grow: 1;
  flex-direction: column;
  padding: 24px 0;
`;

export const AlertInfoContent = styled.div`
  font-family: Roboto;
  font-size: 16px;
  font-weight: 500;
  line-height: 1.38;
  letter-spacing: normal;
  color: #405f71;
`;
export const AlertAdditionalInfoContent = styled.div`
  font-family: Roboto;
  font-size: 16px;
  font-weight: normal;
  line-height: 1.38;
  letter-spacing: 0px;
  color: #587383;
  padding-top: 4px;
`;

export const AlertIcon = styled.div`
  display: flex;
  flex-grow: 0;
  align-items: center;
  padding: 24px;
`;
