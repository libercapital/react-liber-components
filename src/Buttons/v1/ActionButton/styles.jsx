import styled from 'styled-components';
import { media } from '../../../Util';


export const Label = styled.div`
  text-align: center;
  width: unset;
  opacity: 1;
  transition: all 300ms  ${props => (props.hover ? 'ease-out' : 'ease-in')};
  white-space: nowrap;
  color: inherit;

  ${media.large`
    width: ${props => (props.hover ? props.size || '0px' : '0px')};
    opacity: ${props => (props.hover ? '1' : '0')};
  `}
  ${media.extraLarge`
    width: ${props => (props.hover ? props.size || '0px' : '0px')};
    opacity: ${props => (props.hover ? '1' : '0')};
  `}
`;

export const Icon = styled.div`
  margin-right: 15px;
  transition: margin 300ms  ${props => (props.hover ? 'ease-out' : 'ease-in')};
  ${media.large`
    margin-right: ${props => (props.hover ? '15px' : '0px')};
  `}
  ${media.extraLarge`
    margin-right: ${props => (props.hover ? '15px' : '0px')};
  `}
`;
