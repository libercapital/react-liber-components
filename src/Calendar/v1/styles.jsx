import styled, { css } from 'styled-components';
import { handleThemeFromObject } from '../../Themes';

import { CardContent } from '../../Card';
import { FaAngleLeft, FaAngleRight } from '../../Icons';
import { Button } from '../../Buttons';
import { media } from '../../Util';

export const Dot = styled.span`
  width: 6px;
  height: 6px;
  background-color: ${props => handleThemeFromObject(props.theme, 'calendar.dot', '#cecece', props.color)};
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.12);
  display: inline-block;
  border-radius: 3px;
  margin-right: 8px;
`;

export const BiggerDot = styled(Dot)`
  width: unset;
  height: 18px;
  width: unset;
  min-width: 18px;
  border-radius: 9px;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  font-size: 14px;
  color: ${props => handleThemeFromObject(props.theme, 'calendar.biggerDot', '#fff')};
  margin-right: 0px;
  margin-bottom: 5px;
  font-weight: bold;
  padding: 0 5px;
  overflow: hidden;
  box-sizing: border-box;
`;

export const FloatingDots = styled.div`
  position: absolute;
  top: 4px;
  left: ${props => (props.today ? '32px' : '27px')};
  width: 18px;
  display: flex;
  overflow: hidden;
  flex-wrap: wrap;
  & ${Dot} {
    margin-bottom: 2px;
    margin-right: 2px;
  }
  ${media.small`
    left: ${props => (props.today ? '36px' : '31px')};
  `}
  ${media.medium`
    left: ${props => (props.today ? '36px' : '31px')};
  `}
  ${media.large`
    left: ${props => (props.today ? '40px' : '35px')};
  `}
  ${media.extraLarge`
    left: ${props => (props.today ? '40px' : '35px')};
  `}
`;

export const DataContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  font-family: inherit;
  font-size: 11px;
  text-align: center;
  color: ${props => handleThemeFromObject(props.theme, 'calendar.dataContainer', '#717171')};
  text-transform: none;
  margin-bottom: 12px;
  width: unset;
  ${media.small`
    font-size: 11px;
  `}
  ${media.medium`
    font-size: 11px;
  `}
  ${media.large`
    width: 65px;
    font-size: 13px;
  `}
  ${media.extraLarge`
    width: 65px;
    font-size: 13px;
  `}
`;

const tooltipPosition = (position) => {
  switch (position) {
    case 'right':
      return css`
        bottom: 30px;
        left: 30px;
      `;
    case 'left':
      return css`
        bottom: 30px;
        right: 30px;
      `;
    case 'middle':
    default:
      return css`
        bottom: 40px;
        left: calc(50% - calc(148px / 2));
      `;
  }
};

export const Tooltip = styled.div`
  position: absolute;
  ${props => tooltipPosition(props.position)}
  font-family: inherit;
  font-size: 12px;
  font-weight: 500;
  width: 148px;
  border-radius: 4px;
  background-color: ${props => handleThemeFromObject(props.theme, 'calendar.tooltip.background', '#ffffff')};
  box-shadow: 0 2px 6px 0 rgba(0, 0, 0, 0.2);
  color: ${props => handleThemeFromObject(props.theme, 'calendar.tooltip.color', '#717171')};
  padding: 8px;
  display: ${props => (props.hide ? 'none!important' : 'none')};
  z-index: 1000;
  box-sizing: border-box;
  & > h6 {
    margin: 0px;
    opacity: 0.75;
    font-family: inherit;
    font-size: 12px;
    font-weight: bold;
    text-align: left;
    color: ${props => handleThemeFromObject(props.theme, 'calendar.tooltip.color', '#717171')};
    text-transform: capitalize;
    text-align: center;
  }
  & > div {
    display: flex;
    justify-content: flex-start;
    align-items: center;
  }
`;

export const Day = styled.div`
  position: relative;
  font-family: ${props => handleThemeFromObject(props.theme, 'calendar.date.today.fontFamily', 'Roboto')};
  font-size: ${props => handleThemeFromObject(props.theme, 'calendar.date.today.fontSize', '14px')};
  width: 28px;
  height: 28px;
  border: 0px;
  margin: 0px;
  padding: 0px;
  border-radius: 20px;
  user-select: none;
  cursor: ${props => (props.pointer && !props.disabled ? 'pointer' : null)};

  background: ${props => (props.today
    ? handleThemeFromObject(props.theme, 'calendar.date.background', 'linear-gradient(to bottom, #00ccff, #009dff)')
    : 'transparent'
  )};
  box-shadow:  ${props => (props.today ? '0 2px 6px 0 rgba(0, 0, 0, 0.05)' : null)};
  font-weight: ${props => (props.today
    ? handleThemeFromObject(props.theme, 'calendar.date.today.fontWeightToday', '700')
    : handleThemeFromObject(props.theme, 'calendar.date.today.fontWeight', '400')
  )};

  color: ${(props) => {
    if (props.today) {
      return handleThemeFromObject(props.theme, 'calendar.date.today.color', '#fff');
    }
    if (props.disabled) {
      return handleThemeFromObject(props.theme, 'calendar.date.disabled', 'rgb(113,113,113, 0.34)');
    }
    return handleThemeFromObject(props.theme, 'calendar.date.default', '#717171');
  }};

  display: flex;
  justify-content: center;
  align-items: center;

  &:hover {
    background-color: rgba(230, 230, 230, 0.5);
    font-weight: ${props => (props.today
    ? handleThemeFromObject(props.theme, 'calendar.date.today.fontWeightToday', '700')
    : handleThemeFromObject(props.theme, 'calendar.date.today.fontWeightHighlight', '500')
  )};
  }
  &:hover > ${Tooltip} {
    display: block;
  }
  &:hover ${FloatingDots} {
    display: ${props => (props.showDot ? null : 'none')};
  }
  ${media.small`
    font-size: 18px;
    width: 36px;
    height: 36px;
  `}
  ${media.large`
    font-size: 20px;
    width: 40px;
    height: 40px;
  `}
`;

export const DayLink = styled.a`
  text-decoration: none;
  position: relative;
  font-family: inherit;
  font-size: 14px;
  width: 28px;
  height: 28px;
  border: 0px;
  margin: 0px;
  padding: 0px;
  border-radius: 20px;
  user-select: none;
  cursor: ${props => (props.pointer && !props.disabled ? 'pointer' : null)};

  background: ${props => (props.today
    ? handleThemeFromObject(
      props.theme,
      'calendar.date.link.background',
      'linear-gradient(to bottom, #00ccff, #009dff)',
    )
    : 'transparent'
  )};
  box-shadow:  ${props => (props.today ? '0 2px 6px 0 rgba(0, 0, 0, 0.05)' : null)};
  font-weight: ${props => (props.today ? 'bold' : '500')};

  color: ${(props) => {
    if (props.today) {
      return handleThemeFromObject(props.theme, 'calendar.date.link.today', '#fff');
    }
    if (props.disabled) {
      return handleThemeFromObject(props.theme, 'calendar.date.link.disabled', 'rgb(113,113,113, 0.34)');
    }
    return handleThemeFromObject(props.theme, 'calendar.date.link.default', '#717171');
  }};

  display: flex;
  justify-content: center;
  align-items: center;


  &:hover {
    background-color: rgba(230, 230, 230, 0.5);
    color: ${(props) => {
    if (props.today) {
      return handleThemeFromObject(props.theme, 'calendar.date.link.today', '#fff');
    }
    if (props.disabled) {
      return handleThemeFromObject(props.theme, 'calendar.date.link.disabled', 'rgb(113,113,113, 0.34)');
    }
    return handleThemeFromObject(props.theme, 'calendar.date.link.default', '#717171');
  }};
  }
  &:hover > ${Tooltip} {
    display: block;
  }
  &:hover ${FloatingDots} {
    display: ${props => (props.showDot ? null : 'none')};
  }
  ${media.small`
    font-size: 18px;
    width: 36px;
    height: 36px;
  `}
  ${media.medium`
    font-size: 18px;
    width: 36px;
    height: 36px;
  `}
  ${media.large`
    font-size: 20px;
    width: 40px;
    height: 40px;
  `}
  ${media.extraLarge`
    font-size: 20px;
    width: 40px;
    height: 40px;
  `}
`;

export const WeekDay = styled.div`
  width: 30px;
  height: 24px;
  font-family: ${props => handleThemeFromObject(props.theme, 'calendar.week.fontFamily', 'Roboto')};
  font-size: ${props => handleThemeFromObject(props.theme, 'calendar.week.fontSize', '12px')};
  font-weight: ${props => handleThemeFromObject(props.theme, 'calendar.week.fontWeight', '700')};
  text-align: center;
  color: ${props => handleThemeFromObject(props.theme, 'calendar.week.color', '#009dff')};
  text-transform: uppercase;

  ${media.small`
    width: 35.1px;
    font-size: ${props => handleThemeFromObject(props.theme, 'calendar.week.fontSizeSmall', '14px')};
  `}
  ${media.large`
    width: 54.1px;
    font-size: ${props => handleThemeFromObject(props.theme, 'calendar.week.fontSizeLarge', '20px')};
  `}
`;

export const Month = styled.div`
  font-family: ${props => handleThemeFromObject(props.theme, 'calendar.month.fontFamily', 'Roboto')};
  font-size: ${props => handleThemeFromObject(props.theme, 'calendar.month.fontSize', '20px')};
  font-weight: ${props => handleThemeFromObject(props.theme, 'calendar.month.fontWeight', '500')};
  text-align: center;
  color: ${props => handleThemeFromObject(props.theme, 'calendar.month,color', '#717171')};
  text-transform: uppercase;
  width: unset;
  flex-grow: 1;
  ${media.large`
    width: 260px;
    flex-grow: 0;
  `}
`;

export const Week = styled(Month)`
  font-size: ${props => handleThemeFromObject(props.theme, 'calendar.month.fontSize', '20px')};
  width: unset;
  flex-grow: 1;

  ${media.large`
    width: 250px;
    flex-grow: 0;
  `}
`;

export const Row = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  min-height: 36px;
  max-height: 36px;
  ${media.small`
    min-height: 40px;
    max-height: 40px;
  `}
  ${media.medium`
    min-height: 40px;
    max-height: 40px;
  `}
  ${media.large`
    min-height: 43px;
    max-height: 43px;
  `}
  ${media.extraLarge`
    min-height: 43px;
    max-height: 43px;
  `}
`;

export const FilterRow = styled(Row)`
  justify-content: flex-start;
  color: ${props => handleThemeFromObject(props.theme, 'calendar.filterRow', '#717171')} !important;
  font-size: 14px;
  font-weight: 500;
  align-items: flex-start;
  min-height: unset;
  max-height: unset;
  ${media.large`
    align-items: center;
  `}
  ${media.extraLarge`
    align-items: center;
  `}
`;

export const FiltersBox = styled(FilterRow)`
  flex-wrap: wrap;
  font-size: 10px;
  & > *:not(:last-child) {
    margin-bottom: 10px;
  }
  ${media.large`
    font-size: 14px;
    flex-wrap: nowrap;
    & > * {
      margin-bottom: 0px !important;
    }
  `}
  ${media.extraLarge`
    font-size: 14px;
    flex-wrap: nowrap;
    & > * {
      margin-bottom: 0px !important;
    }
  `}
`;

export const FilterLabel = styled.div`
  font-family: inherit;
  font-size: 14px;
  font-weight: 900;
  text-align: center;
  color: ${props => handleThemeFromObject(props.theme, 'calendar.filterLabel', '#717171')};
  margin-right: 20px;
`;

export const Cell = styled.div`
  flex-grow: 1;
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const CalendarBody = styled(CardContent)`
  flex-grow: 1;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: ${props => (props.week ? '0px' : '10px')};
`;

export const Left = styled(FaAngleLeft)`
  width: 24px;
  height: 24px;
  fill: ${props => handleThemeFromObject(props.theme, 'calendar.left', '#009dff')};
`;

export const Right = styled(FaAngleRight)`
  width: 24px;
  height: 24px;
  fill: ${props => handleThemeFromObject(props.theme, 'calendar.right', '#009dff')};
`;

export const NavButton = styled(Button)`
  background-color: transparent;
  border-radius: 50%;
  padding: 0px;
  width: 40px;
  height: 40px;
  &:hover {
    background-color: ${props => (props.disabled ? null : 'rgba(230, 230, 230, 0.5)')};
  }
  margin: 0px;
  & > svg{
    fill: ${props => (props.disabled ? 'rgba(230, 230, 230, 0.5)' : null)};
  }
  ${media.small`
    margin: 0 15px;
  `}
  ${media.medium`
    margin: 0 15px;
  `}
  ${media.large`
    margin: 0 10px;
  `}
  ${media.extraLarge`
    margin: 0 10px;
  `}
`;

export const WeekBody = styled.div`
  flex-grow: 1;
  display: flex;
  justify-content: space-evenly;
  width: 100%;
`;

export const WeekDayValue = styled.div`
  display: inline-flex;
  flex-direction: column;
  font-family: inherit;
  font-size: 18px;
  width: 47px;
  font-weight: 500;
  margin-bottom: 12px;
  color: ${(props) => {
    if (props.today) {
      return handleThemeFromObject(props.theme, 'calendar.date.today', '#009dff');
    }
    if (props.disabled) {
      return handleThemeFromObject(props.theme, 'calendar.date.disabled', 'rgb(113,113,113, 0.34)');
    }
    return handleThemeFromObject(props.theme, 'calendar.date.default', '#717171');
  }};
  align-items: center;
  text-transform: uppercase;
  ${media.large`
    font-size: 20px;
  `}
  ${media.extraLarge`
    font-size: 20px;
  `}
`;

export const WeekCell = styled.div`
  flex-basis: 20%;
  display: flex;
  justify-content: flex-start;
  align-items: center;
  flex-direction: column;
  border-width: 2px;
  border-style: solid;
  min-height: 300px;
  border-image:
    linear-gradient(to bottom, rgba(230, 230, 230, 0.3), rgba(230, 230, 230, 0)) 1 100%;
  &:last-child {
    border: 0px;
  }

  padding-top: 5px;
  cursor: ${props => (props.pointer && !props.disabled ? 'pointer' : null)};
  border-top: 0px;
  border-left: 0px;
  &:hover {
    background-image: linear-gradient(to bottom, rgba(230, 230, 230, 0.3), rgba(230, 230, 230, 0));
  }
  &:hover > ${WeekDayValue} {
    font-weight: 900;
  }
  &:hover ${DataContainer} {
    font-weight: 900;
  }
`;

export const WeekCellLink = styled.a`
  text-decoration: none;
  flex-basis: 20%;
  display: flex;
  justify-content: flex-start;
  align-items: center;
  flex-direction: column;
  border-width: 2px;
  border-style: solid;
  min-height: 300px;
  border-image:
    linear-gradient(to bottom, rgba(230, 230, 230, 0.3), rgba(230, 230, 230, 0)) 1 100%;
  &:last-child {
    border: 0px;
  }

  padding-top: 5px;
  cursor: ${props => (props.pointer && !props.disabled ? 'pointer' : null)};
  border-top: 0px;
  border-left: 0px;
  &:hover {
    background-image: linear-gradient(to bottom, rgba(230, 230, 230, 0.3), rgba(230, 230, 230, 0));
  }
  &:hover > ${WeekDayValue} {
    font-weight: 900;
  }
  &:hover ${DataContainer} {
    font-weight: 900;
  }
`;

export const WeekNav = styled(Row)`
  padding: 15px 0px;
  border-bottom: 2px solid rgba(230, 230, 230, 0.3);
`;
