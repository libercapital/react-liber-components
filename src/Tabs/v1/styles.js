import styled from 'styled-components';
import { handleThemeFromObject } from '../../Themes';
import { handleColor } from '../../Util';

export const Label = styled.label`
  font-family: ${props => handleThemeFromObject(props.theme, 'tabs.label.fontFamily', 'Roboto')};
  font-size: ${props => handleThemeFromObject(props.theme, 'tabs.label.fontSize', '20px')};
  font-weight: ${props => handleThemeFromObject(props.theme, 'tabs.label.fontWeight', '500')};
  text-transform: uppercase;
  background-color: ${props =>
    handleThemeFromObject(
      props.theme,
      'tabs.label.background',
      'transparent',
      props.labelBackgroundColor,
    )};
  color: ${props =>
    props.active
      ? handleThemeFromObject(props.theme, 'tabs.label.active', '#009dff', props.labelActiveColor)
      : handleThemeFromObject(props.theme, 'tabs.label.common', '#717171', props.labelCommonColor)};
  border-bottom: 1px solid
    ${props =>
      handleThemeFromObject(
        props.theme,
        'tabs.label.border',
        'transparent',
        props.labelBorderColor,
      )};
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  cursor: pointer;
  flex-grow: 1;
  flex-basis: 50%;
  margin-bottom: 0px;

  & > input[type='radio'] {
    display: none;
  }

  &:hover {
    color: ${props =>
      handleThemeFromObject(props.theme, 'tabs.label.active', '#009dff', props.labelActiveColor)};
  }
`;

export const BasicHeader = styled.div`
  display: flex;
  height: 100%;
  position: relative;
`;
export const Header = styled(BasicHeader)`
  height: 60px;
  align-items: center;
  width: 100%;
`;

export const Slide = styled.div`
  background: ${props =>
    handleThemeFromObject(
      props.theme,
      'tabs.slider',
      '#009dff',
      props.color ? handleColor(props.color) : null,
    )};
  width: ${props => 100 / props.number}%;
  height: 4px;
  position: absolute;
  left: ${props => (100 / props.number) * props.current}%;
  top: calc(100% - 2px);
  transition: left 0.3s ease-out;
  border: 0px;
`;
