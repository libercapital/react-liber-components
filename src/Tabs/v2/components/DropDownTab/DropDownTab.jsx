import React from 'react';
import PropTypes from 'prop-types';
import ClickOutside from '../../../../Util/ClickOutside';
import { Tab, Content } from './DropDownTab.styles';

const toogleContent = (open, setOpen) => () => {
  setOpen(!open);
};

const closeContent = setOpen => () => {
  setOpen(false);
};

const handleSelection = setCurrent => event => {
  const { target } = event;
  const label = target.getAttribute('label');
  if (label) {
    setCurrent(label);
  }
};

const DropDownTab = ({ active, minWidth, children: childrenProp, strongBorder, label }) => {
  const children = React.Children.toArray(childrenProp);
  if (!children || children.length < 2) {
    throw new Error('You need at least two children, dumbass!');
  }
  const [open, setOpen] = React.useState(false);
  return (
    <Tab
      as="div"
      active={active}
      onClick={toogleContent(open, setOpen)}
      minWidth={minWidth}
      strongBorder={strongBorder}
    >
      <ClickOutside onOutsideClick={closeContent(setOpen)}>
        {label}
        <Content open={open} strongBorder={strongBorder}>
          {children}
        </Content>
      </ClickOutside>
    </Tab>
  );
};

DropDownTab.propTypes = {
  children: PropTypes.arrayOf(PropTypes.element).isRequired,
  active: PropTypes.bool,
  minWidth: PropTypes.number,
  strongBorder: PropTypes.bool,
  label: PropTypes.node,
};

DropDownTab.defaultProps = {
  active: false,
  minWidth: undefined,
  strongBorder: false,
  label: 'Opções',
};

DropDownTab.displayName = 'DropDownTab';

export default DropDownTab;
