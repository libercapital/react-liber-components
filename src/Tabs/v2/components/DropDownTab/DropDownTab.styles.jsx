import styled, { css } from 'styled-components';
import { CenterTab, ChildContainer } from '../SimpleTab/SimpleTab.styles';
import { handleThemeFromObject } from '../../../../Themes';

const getTheme = (key, fallback) => {
  return css`
    ${({ theme }) => handleThemeFromObject(theme, key, fallback)}
  `;
};

export const Tab = styled(CenterTab)`
  position: relative;
  ${({ minWidth }) =>
    minWidth &&
    css`
      min-width: ${minWidth}px;
    `}
  &:after {
    content: '';
    position: absolute;
    top: calc(50% - 9px);
    right: 11px;
    width: 0px;
    height: 0px;
    border-top: 5px solid ${getTheme('colors.petroleum', '#4b6f85')};
    border-right: 5px solid transparent;
    border-bottom: 5px solid transparent;
    border-left: 5px solid transparent;
    transform: scale(0.75);
    margin: 7px;
  }
`;

const toogleView = ({ open }) => {
  if (open) {
    return css`
      opacity: 1;
      transform: translateY(0);
    `;
  }
  return css`
    opacity: 0;
    transform: translateY(-15px);
    pointer-events: none;
  `;
};

export const Content = styled.div`
  position: absolute;
  top: calc(100% + 1px);
  left: -1px;
  border: solid 1px ${({ strongBorder }) =>
    strongBorder ? getTheme('colors.darkBorder', '#cdd7df') : getTheme('colors.grey5', '#ecf0f3')};
  border-top: 0px;
  border-radius: 0px 0px 4px 4px;
  background-color: #fff;
  width: 100%;
  transition: all 250ms cubic-bezier(0.23,1,0.32,1) 0ms;
  z-index: 300;
  ${toogleView}

  & > ${CenterTab} {
    border: 0px;
    background-color: transparent;
    width: 100%;
    height: 36px;
  }
  & ${ChildContainer} {
    pointer-events: none;
  }
`;
