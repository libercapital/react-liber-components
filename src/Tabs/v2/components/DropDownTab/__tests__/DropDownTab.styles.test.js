import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import * as styles from '../DropDownTab.styles';

describe('Testing DropDownTab Styles', () => {
  configure({ adapter: new Adapter() });
  const { Content, ...uncontrolled } = styles;
  describe('Uncontrolled Styles', () => {
    Object.keys(uncontrolled).forEach(style => {
      it(`renders ${style} correctly`, () => {
        const App = uncontrolled[style];
        const wrapper = shallow(<App />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });
    });
  });
  describe('Controlled Styles', () => {
    describe('Content', () => {
      it(`renders correctly without props`, () => {
        const wrapper = shallow(<Content />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });
      it(`renders correctly with open=false`, () => {
        const wrapper = shallow(<Content open={false} />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });
      it(`renders correctly with open=true`, () => {
        const wrapper = shallow(<Content open />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });
    });
  });
});
