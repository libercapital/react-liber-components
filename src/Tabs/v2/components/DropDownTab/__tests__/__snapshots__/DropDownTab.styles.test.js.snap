// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing DropDownTab Styles Controlled Styles Content renders correctly with open=false 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "DropDownTabstyles__Content-sc-1d7lgdx-1",
        "isStatic": false,
        "rules": Array [
          "position:absolute;top:calc(100% + 1px);left:-1px;border:solid 1px ",
          [Function],
          ";border-top:0px;border-radius:0px 0px 4px 4px;background-color:#fff;width:100%;transition:all 250ms cubic-bezier(0.23,1,0.32,1) 0ms;z-index:300;",
          [Function],
          " & > ",
          ".SimpleTabstyles__CenterTab-sc-17by6iu-1",
          "{border:0px;background-color:transparent;width:100%;height:36px;}& ",
          ".SimpleTabstyles__ChildContainer-sc-17by6iu-3",
          "{pointer-events:none;}",
        ],
      },
      "displayName": "DropDownTabstyles__Content",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "DropDownTabstyles__Content-sc-1d7lgdx-1",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  open={false}
/>
`;

exports[`Testing DropDownTab Styles Controlled Styles Content renders correctly with open=true 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "DropDownTabstyles__Content-sc-1d7lgdx-1",
        "isStatic": false,
        "rules": Array [
          "position:absolute;top:calc(100% + 1px);left:-1px;border:solid 1px ",
          [Function],
          ";border-top:0px;border-radius:0px 0px 4px 4px;background-color:#fff;width:100%;transition:all 250ms cubic-bezier(0.23,1,0.32,1) 0ms;z-index:300;",
          [Function],
          " & > ",
          ".SimpleTabstyles__CenterTab-sc-17by6iu-1",
          "{border:0px;background-color:transparent;width:100%;height:36px;}& ",
          ".SimpleTabstyles__ChildContainer-sc-17by6iu-3",
          "{pointer-events:none;}",
        ],
      },
      "displayName": "DropDownTabstyles__Content",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "DropDownTabstyles__Content-sc-1d7lgdx-1",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  open={true}
/>
`;

exports[`Testing DropDownTab Styles Controlled Styles Content renders correctly without props 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "DropDownTabstyles__Content-sc-1d7lgdx-1",
        "isStatic": false,
        "rules": Array [
          "position:absolute;top:calc(100% + 1px);left:-1px;border:solid 1px ",
          [Function],
          ";border-top:0px;border-radius:0px 0px 4px 4px;background-color:#fff;width:100%;transition:all 250ms cubic-bezier(0.23,1,0.32,1) 0ms;z-index:300;",
          [Function],
          " & > ",
          ".SimpleTabstyles__CenterTab-sc-17by6iu-1",
          "{border:0px;background-color:transparent;width:100%;height:36px;}& ",
          ".SimpleTabstyles__ChildContainer-sc-17by6iu-3",
          "{pointer-events:none;}",
        ],
      },
      "displayName": "DropDownTabstyles__Content",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "DropDownTabstyles__Content-sc-1d7lgdx-1",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing DropDownTab Styles Uncontrolled Styles renders Tab correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "DropDownTabstyles__Tab-sc-1d7lgdx-0",
        "isStatic": false,
        "rules": Array [
          "--icon-size:16px;--border-color:",
          [Function],
          ";min-height:48px;flex-grow:1;display:flex;justify-content:center;align-items:center;flex-basis:0;font-family:",
          [Function],
          ";",
          [Function],
          " text-decoration:none;border:0px;border-top:1px solid var(--border-color);border-right:1px solid var(--border-color);&:first-of-type{border-radius:4px 0 0 0;border-left:1px solid var(--border-color);}&:last-of-type{border-radius:0 4px 0 0;}",
          [Function],
          " ",
          [Function],
          " &:focus{outline:0;}cursor:pointer;transition:all 450ms cubic-bezier(0.23,1,0.32,1) 0ms;white-space:nowrap;",
          "position:relative;",
          [Function],
          " &:after{content:'';position:absolute;top:calc(50% - 9px);right:11px;width:0px;height:0px;border-top:5px solid ",
          [Function],
          ";border-right:5px solid transparent;border-bottom:5px solid transparent;border-left:5px solid transparent;transform:scale(0.75);margin:7px;}",
        ],
      },
      "displayName": "DropDownTabstyles__Tab",
      "foldedComponentIds": Array [
        "SimpleTabstyles__CenterTab-sc-17by6iu-1",
      ],
      "render": [Function],
      "styledComponentId": "DropDownTabstyles__Tab-sc-1d7lgdx-0",
      "target": "a",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
